compiler/aarch64/codegen/GenerateInstructions.cpp

```diff
@@ -125,6 +125,25 @@ TR::Instruction *generateCompareBranchInstruction(TR::CodeGenerator *cg, TR::Ins
    return new (cg->trHeapMemory()) TR::ARM64CompareBranchInstruction(op, node, sreg, sym, cond, cg);
    }
 
+TR::Instruction *generateCbzOrCbnzInstruction(TR::CodeGenerator *cg, TR::Node *node, TR::Register *sreg,
+   TR::LabelSymbol *sym, bool is64bit, TR::Instruction *preced)
+   {
+   TR::InstOpCode::Mnemonic op;
+   int64_t value = is64bit ? node->getFirstChild()->getLongInt() : node->getFirstChild()->getInt();
```

Comment:
You mean to check the second child here, don't you?

Reply:
This function is called  when the second child is zero.
I am checking the first child value to decide, if the op should be cbz or cbnz.
Did I understand anything wrong?

Reply:
@0xdaryl Can you please confirm, will it be second child here?

Reply:
Sorry, I had this wrong too.   You've already checked the second child for the `A != 0` or `A == 0` pattern prior to calling this function, so checking the first child isn't going to tell you anything (both children won't be constants because the simplifier would have folded this whole comparison away).  Here you need to check whether it's a compare equals or not to distinguish the cases.  i.e., `TR::ILOpCode::isEqualCmp(node->getOpCode().getOpCodeValue())` will tell you that.

For safety, you may wish to add a `TR_FATAL_ASSERT` that the node is either `isEqualCmp` or `isNotEqualCmp`.

Reply:
Necessary changes applied.

Reply:
@knn-k Here is the suggestion about using the `TR::ILOpCode::isEqualCmp(node->getOpCode().getOpCodeValue())`. I think I could use the `(cc == TR::CC_EQ)` also. According to my understanding both are same.